pkg/probe/http/http.go

```diff
@@ -90,7 +116,7 @@ type GetHTTPInterface interface {
 // If the HTTP response code is successful (i.e. 400 > code >= 200), it returns Success.
 // If the HTTP response code is unsuccessful or HTTP communication fails, it returns Failure.
 // This is exported because some other packages may want to do direct HTTP probes.
-func DoHTTPProbe(req *http.Request, client GetHTTPInterface) (probe.Result, string, error) {
+var DoHTTPProbe = func(req *http.Request, client GetHTTPInterface) (probe.Result, string, error) {
```

Comment:
Can you explain why is this change needed?

Reply:
I changed this for the convenience of running the benchmark.  In the benchmark, this func variable is changed to a function always return true to mock the behavior of doing an actual http probe.  If this pr is to be merged, I think this change and the associated benchmark code is not necessary to be included.